hydra_base/lib/project.py

```diff
@@ -233,21 +234,39 @@ def to_named_tuple(obj, visited_children=None, back_relationships=None, levels=N
 
 def get_projects(uid,**kwargs):
     """
-        get a project complexmodel
+        Get all the projects owned by the specified user.
+        These include projects created by the user, but also ones shared with the user.
+        For shared projects, only include networks in those projects which are accessible to the user.
     """
     req_user_id = kwargs.get('user_id')
 
-    #Potentially join this with an rs of projects
-    #where no owner exists?
-
-    projects = db.DBSession.query(Project).join(ProjectOwner).filter(Project.status=='A', ProjectOwner.user_id==uid).options(joinedload_all('networks')).order_by('id').all()
-    for project in projects:
-        project.check_read_permission(req_user_id)
+    ##Don't load the project's networks. Load them separately, as the networks
+    #must be checked individually for ownership
+    projects_i = db.DBSession.query(Project).join(ProjectOwner)\
+                                                 .filter(Project.status=='A',
+                                                        or_(ProjectOwner.user_id==uid,
+                                                           Project.created_by==uid))\
+                                                 .options(noload('networks'))\
+                                                 .order_by('id').all()
+
+    #Load each 
+    projects_j = []
+    for project_i in projects_i:
+        #Ensure the requesting user is allowed to see the project
+        project_i.check_read_permission(req_user_id)
```

Comment:
Is there no equivalent `check_read_permission` on a `Network`?

Reply:
This will raise an exception if it fails the check. Should that just skip this project or allow the exception propagate (this is current behaviour)?

Reply:
propagating the excceptiopn is how it currently works. Note that the check refers to the requesting user, not the one for whom the request is being made ("get me projects for uid", but requested by req_user_id)

Reply:
I've added a check_read_permission call on each sub network of each project.

Reply:
OK fair enough.